### PR TITLE
🌱 feat: add k3s providers from the k3s.io project to clusterctl

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

This adds the bootstrap and controlplane providers for k3s to clusterctl.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cmd/clusterctl/client/config/providers_client.go [hunk 1]
```diff
@@ -368,6 +375,11 @@ func (p *providersClient) defaults() []Provider {
 			url:          "https://github.com/k0sproject/k0smotron/releases/latest/control-plane-components.yaml",
 			providerType: clusterctlv1.ControlPlaneProviderType,
 		},
+		&provider{
+			name:         K3sBootstrapProviderName,
```
Comment (furkatgofurov7) [inline cmd/clusterctl/client/config/providers_client.go:379]:
```suggestion
			name:         K3sControlPlaneProviderName,
```

Comment (richardcase) [reply cmd/clusterctl/client/config/providers_client.go]: Thanks @furkatgofurov7 , changed.

**Comment by sbueringer** [comment]
@richardcase Are you still interested in pursuing this?

**Comment by richardcase** [comment]
> @richardcase Are you still interested in pursuing this?

Yes for sure. Sorry i lost track of this. I will get it updated.

File: cmd/clusterctl/client/config/providers_client.go [hunk 2]
```diff
@@ -76,6 +76,7 @@ const (
 	KubeKeyK3sBootstrapProviderName        = "kubekey-k3s"
 	RKE2BootstrapProviderName              = "rke2"
 	K0smotronBootstrapProviderName         = "k0sproject-k0smotron"
+	K3sBootstrapProviderName               = "k3s"
```
[the remark this answers is not in the document]
Comment (richardcase) [reply cmd/clusterctl/client/config/providers_client.go]: I have changed this.

**Comment by sbueringer** [comment]
@richardcase Looks like the test needs a fix

**Comment by richardcase** [comment]
> @richardcase Looks like the test needs a fix

Yeah, i started to look at the fix and completely distracted.....again. I will get it fixed asap. Thanks for the follow up @sbueringer

**Comment by Jont828** [comment]
Based on this test output it appears that it's running a diff where it's still expecting the name to be kubeadm.

```
{Failed  === RUN   Test_runGetRepositories/prints_output
    config_repositories_test.go:55: 
        Expected
            <string>:   (
              	"""
              	... // 11 identical lines
              	  URL: https://github.com/k0sproject/k0smotron/releases/latest/
              	- File: bootstrap-components.yaml
            - 	  Name: kubeadm
            + 	  Name: k3s-k3s
              	  ProviderType: BootstrapProvider
            - 	  URL: https://github.com/kubernetes-sigs/cluster-api/releases/latest/
            + 	  URL: https://github.com/k3s-io/cluster-api-k3s/releases/latest/
              	- File: bootstrap-components.yaml
            - 	  Name: kubekey-k3s
            + 	  Name: kubeadm
              	  ProviderType: BootstrapProvider
            + 	  URL: https://github.com/kubernetes-sigs/cluster-api/releases/latest/
            + 	- File: bootstrap-components.yaml
            + 	  Name: kubekey-k3s
            + 	  ProviderType: BootstrapProvider
              	  URL: https://github.com/kubesphere/kubekey/releases/latest/
              	- File: bootstrap-components.yaml
              	... // 13 identical lines
              	  ProviderType: BootstrapProvider
              	  URL: https://github.com/siderolabs/cluster-api-bootstrap-provider-talos/releases/latest/
            + 	- File: control-plane-components.yaml
            + 	  Name: k0sproject-k0smotron
            + 	  ProviderType: ControlPlaneProvider
            + 	  URL: https://github.com/k0sproject/k0smotron/releases/latest/
              	- File: control-plane-components.yaml
            - 	  Name: k0sproject-k0smotron
            + 	  Name: k3s-k3s
              	  ProviderType: ControlPlaneProvider
            - 	  URL: https://github.com/k0sproject/k0smotron/releases/latest/
            + 	  URL: https://github.com/k3s-io/cluster-api-k3s/releases/latest/
              	- File: control-plane-components.yaml
              	  Name: kamaji
              	... // 159 identical lines
              	"""
              )
            
        to be empty
--- FAIL: Test_runGetRepositories/prints_output (0.01s)
}
```

**Comment by sbueringer** [comment]
/close 
Due to inactivity 
Feel free to reopen and address the findings

**Pull request closed without merging**